src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -796,13 +793,21 @@ cldnn::network::ptr primitive_inst::get_unfused_subgraph() {
                 // which doesn't exist anymore in the graph
                 // Thus we update dependency name used dependencies idx stored in fused descriptor.
                 if (std::find(dep_ids.begin(), dep_ids.end(), in) == dep_ids.end()) {
-                    size_t dep_id = fd.dep_start_idx + i;
+                    size_t dep_id = fd.dep_start_idx + i - 1;
                     in = _node.get_dependency(dep_id).id();
                 }
             }
             t.add_primitive(prim);
             dep_ids.push_back(prim->id);
         }
+        // Samely, need to update dependency of the current fused nodes' input primitive ids with those in the current program
+        auto prim_of_fused_node = std::const_pointer_cast<primitive>(_impl_params->desc);
```

Comment:
If you've found an issue with unfusion, please add unit test for that case

Reply:
Added